methods.go

```diff
@@ -205,12 +207,73 @@ func decodeTorrentsStatusResponse(resp *DelugeResponse) (map[string]*TorrentStat
 	return result, nil
 }
 
+// TorrentError is a tuple of a torrent and an error message, returned by
+// methods that manipulate many torrents at once.
+type TorrentError struct {
+	// ID is the hash of the torrent that experienced an error
+	ID string
+	// Message is a string message
+	Message string
+}
+
+// Error satisfies the error interface
+func (t TorrentError) Error() string {
+	return fmt.Sprintf("<%s>: '%s'", t.ID, t.Message)
+}
+
+// RemoveTorrents tries to remove multiple torrents at once.
+// If `rmFiles` is set it also tries to delete all downloaded data for the
+// specified torrents.
+// If errors were encountered, the returned list will be a list of
+// TorrentErrors.
+// On success, an empty list of errors is returned.
+func (c *Client) RemoveTorrents(ids []string, rmFiles bool) ([]TorrentError, error) {
+	var args rencode.List
+	args.Add(sliceToRencodeList(ids), rmFiles)
+
+	resp, err := c.rpc("core.remove_torrents", args, rencode.Dictionary{})
+	if err != nil {
+		return nil, err
+	}
+	if resp.IsError() {
+		return nil, resp.RPCError
+	}
+
+	vals := resp.returnValue.Values()
+	if len(vals) == 0 {
+		return nil, ErrInvalidReturnValue
+	}
+	failedList := vals[0].(rencode.List)
+
+	var torrentErrors []TorrentError
+
+	// Iterate through the list of errors that have occured, and
+	// convert each of them into a more typesafe format.
+	for _, e := range failedList.Values() {
+		failedEntry, ok := e.(rencode.List)
+		if !ok {
+			continue
```

Comment:
What is the loop ignoring here?

Reply:
Now returns an Error on unexpected output, as this should not happen unnoticed.